packages/migrate/migrations/self-closing-tags/migrate.js

```diff
@@ -0,0 +1,117 @@
+const VoidElements = [
+	'area',
+	'base',
+	'br',
+	'col',
+	'embed',
+	'hr',
+	'img',
+	'input',
+	'keygen',
+	'link',
+	'menuitem',
+	'meta',
+	'param',
+	'source',
+	'track',
+	'wbr'
+];
+
+const SVGElements = [
+	'altGlyph',
+	'altGlyphDef',
+	'altGlyphItem',
+	'animate',
+	'animateColor',
+	'animateMotion',
+	'animateTransform',
+	'circle',
+	'clipPath',
+	'color-profile',
+	'cursor',
+	'defs',
+	'desc',
+	'discard',
+	'ellipse',
+	'feBlend',
+	'feColorMatrix',
+	'feComponentTransfer',
+	'feComposite',
+	'feConvolveMatrix',
+	'feDiffuseLighting',
+	'feDisplacementMap',
+	'feDistantLight',
+	'feDropShadow',
+	'feFlood',
+	'feFuncA',
+	'feFuncB',
+	'feFuncG',
+	'feFuncR',
+	'feGaussianBlur',
+	'feImage',
+	'feMerge',
+	'feMergeNode',
+	'feMorphology',
+	'feOffset',
+	'fePointLight',
+	'feSpecularLighting',
+	'feSpotLight',
+	'feTile',
+	'feTurbulence',
+	'filter',
+	'font',
+	'font-face',
+	'font-face-format',
+	'font-face-name',
+	'font-face-src',
+	'font-face-uri',
+	'foreignObject',
+	'g',
+	'glyph',
+	'glyphRef',
+	'hatch',
+	'hatchpath',
+	'hkern',
+	'image',
+	'line',
+	'linearGradient',
+	'marker',
+	'mask',
+	'mesh',
+	'meshgradient',
+	'meshpatch',
+	'meshrow',
+	'metadata',
+	'missing-glyph',
+	'mpath',
+	'path',
+	'pattern',
+	'polygon',
+	'polyline',
+	'radialGradient',
+	'rect',
+	'set',
+	'solidcolor',
+	'stop',
+	'svg',
+	'switch',
+	'symbol',
+	'text',
+	'textPath',
+	'tref',
+	'tspan',
+	'unknown',
+	'use',
+	'view',
+	'vkern'
+];
+
+/**
+ * @param {string} code
+ */
+export function remove_self_closing_tags(code) {
+	return code.replace(/<([a-z-]+)(\s[^>]+?)? ?\/>/g, (match, name, attributes = '') => {
```

Comment:
This is too broad:
- doesn't leave foreign namespace alone
- forces slot expansion even if those tags are not really real html tags, Svelte hijacks them to mean something different and as such self closing should be allowed. If we don't do that then I think the migration will produce a lot less noise for a piece of syntax that is deprecated anyway

Reply:
What foreign namespaces are we talking about? MathML? I'm not aware of any MathML elements that make sense without child content. Are people using other namespaces in Svelte components? The worst case scenario is we replace `<valid />` with `<valid></valid>`. It would overcomplicate things to an absurd degree if we tried to avoid that.

Similarly, to distinguish between real `<slot>` elements and fake ones, we would need to determine whether a given `.svelte` component was intended to be compiled as a custom element, which... I don't know, I guess we would need to traverse the project looking for a `svelte.config.js` file (and miss cases where config was set directly via a bundler plugin) or something? And then we'd have `<slot />` in some places and `<slot></slot>` in others and we'd need to explain why that was the case.

Seems like a bad use of time and energy!

Reply:
- I'm talking about the `namespace=foreign` compiler option, which is used for Svelte Native etc. . Changing their code is just unnecessary.
- For custom elements we just need to see if there's a customElement or tag option on `svelte:options`.
- Finding a svelte.config.js is really easy and was done in various migration scripts before.
- avoiding the slot thing will make the whole migration a lot less noisy and be accepted by more people.

Getting a migration script correct is a good spend of our time!